### PR TITLE
libvirt-e2e: create auth.json file

### PR DESCRIPTION
**Pull request by mkulke**
The auth.json file is not provisioned for the authenticated registry tests in the libvirt test suite. This adds the file and adjusts the kustomize configuration to use it.

drive-by fix: log formatting

### CODE REVIEW

**Review by liudalibj — APPROVED** [review]
LGTM, thanks for the fix!

**Comment by mkulke** [comment]
ok, that seems to fix the libvirt tests

File: src/cloud-api-adaptor/test/e2e/assessment_runner.go [hunk 1]
```diff
@@ -276,15 +276,15 @@ func (tc *TestCase) Run() {
 						t.Logf("Expected Pod State: %v", tc.podState)
 						yamlData, err := yaml.Marshal(pod.Status)
 						if err != nil {
-							fmt.Println("Error marshaling pod.Status to YAML: ", err.Error())
+							t.Logf("Error marshaling pod.Status to YAML: ", err.Error())
```
Comment (bpradipt) [inline src/cloud-api-adaptor/test/e2e/assessment_runner.go:279]: needs fixing ^

File: src/cloud-api-adaptor/test/provisioner/libvirt/provision_common.go [hunk 2]
```diff
@@ -337,6 +345,16 @@ func (lio *LibvirtInstallOverlay) Edit(ctx context.Context, cfg *envconf.Config,
 		}
 	}
 
+	if cred := os.Getenv("REGISTRY_CREDENTIAL_ENCODED"); cred != "" {
```
Comment (wainersm) [inline src/cloud-api-adaptor/test/provisioner/libvirt/provision_common.go:348]: Hi @mkulke , I'm about to send a PR with fixes for AWS and I was wondering how to create the auth.json. So likely I will reuse your work :)

File: src/cloud-api-adaptor/test/e2e/assessment_runner.go (hunk 1 above)
Comment (wainersm) [reply src/cloud-api-adaptor/test/e2e/assessment_runner.go:279]: Also the commit message is a little ugly.

Comment (bpradipt) [inline src/cloud-api-adaptor/test/e2e/assessment_runner.go:279]: Looks like it didn't get refreshed ?

**Review by bpradipt — APPROVED** [review]
/lgtm
Thanks @mkulke

**Pull request merged**